### PR TITLE
issue: 964360 Improve Page Fault event on mlx5_send call

### PR DESCRIPTION
**Pull request by igor-ivanov**
These changes designed to eliminate page fault event during
qp_mgr::mlx5_send() call.

@OphirMunk please review. I hope that these changes can improve page fault issue in this specific function.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/916/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/771/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/919/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/774/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
@igor-ivanov please issue this same PR on experimental branch as well

**Comment by OphirMunk** [comment]
@igor-ivanov please fix Jenkins failures

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/925/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/930/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/940/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/943/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/784/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/945/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/949/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/952/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/954/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/955/ for details (Mellanox internal link).

File: src/vma/event/delta_timer.cpp [hunk 1]
```diff
@@ -73,13 +73,14 @@ timer::~timer()
 	NOT_IN_USE(iter);
 	NOT_IN_USE(to_free);
 	return;
-#endif // DEFINED_VMAPOLL		
+#else
```
Comment (OphirMunk) [inline src/vma/event/delta_timer.cpp]: @igor-ivanov Is this fix related to page faults? If not - would you consider a separate commit?

File: src/vma/sock/sock-redirect.cpp [hunk 2]
```diff
@@ -541,15 +541,18 @@ extern "C"
 int vma_dump_fd_stats(int fd, int log_level)
 {
 #ifdef DEFINED_VMAPOLL
+NOT_IN_USE(fd);
+NOT_IN_USE(log_level);
 	return 0;
-#endif // DEFINED_VMAPOLL
+#else
```
Comment (OphirMunk) [inline src/vma/sock/sock-redirect.cpp]: @igor-ivanov Is this fix related to page faults? If not - would you consider a separate commit?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/957/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/958/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/787/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/966/ for details (Mellanox internal link).

File: src/vma/sock/sock-redirect.cpp (hunk 2 above)
Comment (igor-ivanov) [inline src/vma/sock/sock-redirect.cpp]: It is fix for Intel Compiler

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/969/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/979/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/982/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/984/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/987/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/992/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1017/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Merged #292

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1067/ for details (Mellanox internal link).
